### DISCUSSION TITLE
How to draw a spiral block/boundary?

### DISCUSSION THREAD

**Discussion by Yasunaoka** [idaholab/moose#27860] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi guys, I recently wanted to draw a spiral p=e^θ（(as shown by the green line on the diagram) or a series of points on that spiral over a 2D circular mesh so that I can specify the initial condition variable c=1 in the region. i have tried a few objects but didn't succeed. Does anyone know how to implement this elegantly please? Any help is much appreciated!
![image](https://github.com/idaholab/moose/assets/147731232/6fc4f75b-0963-4cca-9c29-b45a17f8423d)

**Comment by GiudGiud** [comment]
Hello

Do you want the mesh to have a spirally shape? Or simply cover part of the domain with a spiral?
You are looking for 2D surface right? Or a 1D spiral curve?

Use this generator
https://mooseframework.inl.gov/source/meshgenerators/ConcentricCircleMeshGenerator.html
Then this one to transform the circles into spirals
https://mooseframework.inl.gov/source/meshgenerators/ParsedNodeTransformGenerator.html
You'll have to figure out the transformation in cartesian coordinates

Guillaume

**Reply by Yasunaoka** [reply]
Sorry, I may have missed the point, I currently do have a circular mesh ```AnnularMeshGenerator``` already:
![image](https://github.com/idaholab/moose/assets/147731232/196b5884-5dee-402f-91d1-e2c12d789bae)
So far, I have gotten the following linear initial condition via ```BoundingBoxIC```
![image](https://github.com/idaholab/moose/assets/147731232/d275dee7-1df4-4fba-91ec-8cd1c4c43e23)
But now I'm trying to get the spiral pattern shown in the green line below, and I'm not finding the right way to do it.
![image](https://github.com/idaholab/moose/assets/147731232/64fd9f36-cf0e-4026-a2b6-072407b4fc16)

**Reply by GiudGiud** [reply, marked as answer]
if you want the mesh to rigorously align with the spiral, you could rotate the mesh (not evenly, with the parsedTransform) first.
then use a ParsedGenerateSideset to select the spiral sideset

If you are OK with a jagged edge, then you could simply use the ParsedGenerateSideset to approximately draw the spiral bit within the mesh

**Reply by Yasunaoka** [reply]
That's exactly what I want! That's great! Thank you so much!
